### PR TITLE
new(async query): new NG doc

### PR DESCRIPTION
**Pull request by zuluecho9**
NR-34814

Work: 
New 'Async query' doc in NerdGraph category
Links to it from Data Plus section and from 'query duration limit' doc.

### CODE REVIEW

File: src/content/docs/apis/nerdgraph/examples/async-queries-nrql-tutorial.mdx
Comment (jmschles) [inline src/content/docs/apis/nerdgraph/examples/async-queries-nrql-tutorial.mdx:26]: Minor formatting tweak: this line needs to be "de-dented" two spaces

File: src/content/docs/apis/nerdgraph/examples/async-queries-nrql-tutorial.mdx
Comment (jmschles) [inline src/content/docs/apis/nerdgraph/examples/async-queries-nrql-tutorial.mdx:14]: I'll defer to you on this, but to be clear: Data Plus isn't a requirement for asynchronous queries, it's just a requirement for query durations that extend all the way out to 10 minutes. To me, this is a little confusing, because it reads like I need Data Plus to run async queries at all. But I also realize from the title that this doc is mainly concerned with 10-minute queries, so maybe it's fine as is.

File: src/content/docs/apis/nerdgraph/examples/async-queries-nrql-tutorial.mdx
Comment (jmschles) [inline src/content/docs/apis/nerdgraph/examples/async-queries-nrql-tutorial.mdx:18]: Probably want to put `queryProgress` in backticks since it's the name of a field.

**Review by jmschles — COMMENTED** [review]
This looks good from an engineering standpoint. I just had a couple small comments.

File: src/content/docs/apis/nerdgraph/examples/async-queries-nrql-tutorial.mdx
Comment (zuluecho9) [reply src/content/docs/apis/nerdgraph/examples/async-queries-nrql-tutorial.mdx:14]: ah i see, right, sorry

**Review by jmschles — DISMISSED** [review]

File: src/content/docs/accounts/accounts-billing/new-relic-one-pricing-billing/data-ingest-billing.mdx
Comment (akristen) [inline src/content/docs/accounts/accounts-billing/new-relic-one-pricing-billing/data-ingest-billing.mdx:39]:
```suggestion
* Increased retention of several important data types for 98 days ([learn more](/docs/data-apis/manage-data/manage-data-retention)).
```

File: src/content/docs/apis/nerdgraph/examples/async-queries-nrql-tutorial.mdx
Comment (akristen) [inline src/content/docs/apis/nerdgraph/examples/async-queries-nrql-tutorial.mdx:14]: Do you want (edited) to be here? Calling it out just in case

File: src/content/docs/apis/nerdgraph/examples/async-queries-nrql-tutorial.mdx
Comment (akristen) [inline src/content/docs/apis/nerdgraph/examples/async-queries-nrql-tutorial.mdx:10]:
```suggestion
Our NerdGraph API can run an asynchronous NRQL query with `async: true`. If you have a query that may take a long time to complete, using an asynchronous NRQL query introduces less risk of query interruption by browser or HTTP connection timeout limits or other issues.
```

Comment (akristen) [reply src/content/docs/apis/nerdgraph/examples/async-queries-nrql-tutorial.mdx:10]: Curious if "other issues" is necessary -- I'd recommend either omitting it entirely unless it prevents interruption from any kind of issue. e.g., does this preserve the ~integrity of a query if there's infrastructure failure, or is only useful for interruptions? Not sure if that nuance makes sense but it might be worth teasing apart at least the genre of issues it reduces risk for

File: src/content/docs/apis/nerdgraph/examples/async-queries-nrql-tutorial.mdx
Comment (akristen) [inline src/content/docs/apis/nerdgraph/examples/async-queries-nrql-tutorial.mdx:18]: I'd add a sentence of exposition because when I got to the second example, I wasn't quite sure what exactly I was looking at until I reread it. Some ideas:

> An asynchronous query will return the results if it completes within the default timeout of 5 seconds. If an async query exceeds 5 seconds, then the query will return a `queryProgress` ID, which you can use to get results or the progress status of that query exceeding 5 seconds. 

It's a little redundant from a language-perspective, but I think being redundant works in the service of absolute clarity. A technical question I have is:

- Can you use the `queryProgress` ID to get partial results? i.e., if I had a stacked query to retrieve 2 discrete sets of data an datum A was retrieved before the code could get datum B, would I be able to use `queryProgress` to return A and get progress on B? Idk if that's just a science fiction example but it occurred to me~ 
- I'm not entirely clear what the `actor.account.nrqlQueryProgress` field is. Can you use `actor.account.nrqlQueryProgress` to do what I was asking about above ^?

File: src/content/docs/apis/nerdgraph/examples/async-queries-nrql-tutorial.mdx
Comment (akristen) [inline src/content/docs/apis/nerdgraph/examples/async-queries-nrql-tutorial.mdx:41]:
```suggestion
If your query exceeds the default timeout, it will return a query ID. Take that ID and run a second query to retrieve the results or progress status for that query. 
```

**Review by akristen — CHANGES_REQUESTED** [review]
Let me know when you need a PR approval after you make some changes! I'll be around :)

File: src/content/docs/apis/nerdgraph/examples/async-queries-nrql-tutorial.mdx
Comment (zuluecho9) [reply src/content/docs/apis/nerdgraph/examples/async-queries-nrql-tutorial.mdx:14]: ha. whoops

**Pull request merged**
